core/schema/src/main/java/org/hawkular/metrics/schema/VersionUtil.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2014-2018 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.hawkular.metrics.schema;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Enumeration;
+import java.util.jar.Manifest;
+
+import com.google.common.base.Strings;
+
+/**
+ * @author jsanda
+ */
+public class VersionUtil {
+
+    public static String getVersion() {
+        try {
+            Enumeration<URL> resources = VersionUtil.class.getClassLoader().getResources("META-INF/MANIFEST.MF");
+            while (resources.hasMoreElements()) {
+                URL resource = resources.nextElement();
+                try (InputStream stream = resource.openStream()) {
+                    Manifest manifest = new Manifest(stream);
+                    String vendorId = manifest.getMainAttributes().getValue("Implementation-Vendor-Id");
+                    String implVersion = manifest.getMainAttributes().getValue("Implementation-Version");
+                    String gitSHA = manifest.getMainAttributes().getValue("Built-From-Git-SHA1");
+
+                    if (isValidManifest(vendorId, implVersion, gitSHA)) {
+                        return implVersion + "+" + gitSHA.substring(0, 10);
```

Comment:
0, 10 ? What's the logic behind this selection? Theoretically git shortens to shortest possible unambiguous SHA1. But that can change. It just happens to be 9 bytes in our current repository.

Reply:
Based on what I read [here](https://git-scm.com/book/en/v2/Git-Tools-Revision-Selection#Short-SHA-1), I think 10 characters is more than sufficient.